From c30d758b98a11ba9f908ba47c204a4847e315283 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Val=C3=A9rio=20Val=C3=A9rio?= <valerio.valerio@jolla.com>
Date: Mon, 11 May 2015 15:12:07 +0300
Subject: [PATCH 19/20] Check if IDLE connection needs to be established after
 login. Fixes JB#28693

Some servers only advertise full capabilities after sucessful login,
so we need to check if IDLE connection needs to be established in such
cases.
---
 src/plugins/messageservices/imap/imapclient.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/plugins/messageservices/imap/imapclient.cpp b/src/plugins/messageservices/imap/imapclient.cpp
index 1840fa79..cf5fc72a 100644
--- a/src/plugins/messageservices/imap/imapclient.cpp
+++ b/src/plugins/messageservices/imap/imapclient.cpp
@@ -872,6 +872,16 @@ void ImapClient::commandTransition(ImapCommand command, OperationStatus status)
                 }
 #endif
             }
+            // After logging in server capabilities reported may change so we need to
+            // check if IDLE is already established, when enabled
+            if (!_waitingForIdle && !_idlesEstablished
+                && _protocol.supportsCapability("IDLE")
+                && !_waitingForIdleFolderIds.isEmpty()
+                && _pushConnectionsReserved) {
+                _waitingForIdle = true;
+                monitor(_waitingForIdleFolderIds);
+                emit updateStatus( tr("Logging in idle connection" ) );
+            }
 
             bool compressCapable(_protocol.capabilities().contains("COMPRESS=DEFLATE", Qt::CaseInsensitive));
             if (!_protocol.encrypted() && QMFALLOWCOMPRESS && compressCapable && !_protocol.compress()) {
-- 
2.17.1

